code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardSchemaData.java

```diff
@@ -0,0 +1,143 @@
+            this.expiryDate =
+                    schemaData.optInt(MessagingConstants.ConsequenceDetailDataKeys.EXPIRY_DATE);
+            this.meta =
+                    JSONUtils.toMap(
+                            schemaData.optJSONObject(
+                                    MessagingConstants.ConsequenceDetailDataKeys.METADATA));
+        } catch (final JSONException exception) {
+            Log.trace(
+                    MessagingConstants.LOG_TAG,
+                    SELF_TAG,
+                    "Exception occurred creating FeedItemSchemaData from json object: %s",
+                    exception.getLocalizedMessage());
+        }
+    }
+
+    @Override
+    public Object getContent() {
+        return content;
+    }
+
+    public ContentType getContentType() {
+        return contentType;
+    }
+
+    public int getPublishedDate() {
+        return publishedDate;
+    }
+
+    public int getExpiryDate() {
+        return expiryDate;
+    }
+
+    @Nullable public Map<String, Object> getMeta() {
+        return meta;
+    }
+
+    @VisibleForTesting
+    static ContentCardSchemaData getEmpty() {
+        return new ContentCardSchemaData(new JSONObject());
+    }
+
+    @Nullable public ContentCard getContentCard() {
+        if (!contentType.equals(ContentType.APPLICATION_JSON)) {
+            return null;
+        }
+
+        try {
+            final Map<String, Object> contentMap = (HashMap<String, Object>) content;
+            final String title =
+                    DataReader.optString(contentMap, MessagingConstants.MessageFeedKeys.TITLE, "");
+            final String body =
+                    DataReader.optString(contentMap, MessagingConstants.MessageFeedKeys.BODY, "");
+            final String imageUrl =
+                    DataReader.optString(
+                            contentMap, MessagingConstants.MessageFeedKeys.IMAGE_URL, "");
+            final String actionUrl =
+                    DataReader.optString(
+                            contentMap, MessagingConstants.MessageFeedKeys.ACTION_URL, "");
+            final String actionTitle =
+                    DataReader.optString(
+                            contentMap, MessagingConstants.MessageFeedKeys.ACTION_TITLE, "");
+            return new ContentCard.Builder(title, body)
+                    .setImageUrl(imageUrl)
+                    .setActionUrl(actionUrl)
+                    .setActionTitle(actionTitle)
+                    .setParent(this)
+                    .build();
+        } catch (final ClassCastException exception) {
+            return null;
+        }
+    }
+
+    /**
+     * Tracks interaction with the given proposition item.
+     *
+     * @param interaction {@link String} describing the interaction.
+     * @param eventType enum of type {@link MessagingEdgeEventType} specifying event type for the
+     *     interaction.
+     */
+    public void track(final String interaction, final MessagingEdgeEventType eventType) {
```

Comment:
Where is this new `track` method supposed to be called from? I only see the `Propositionitem` track being called everywhere in the code as well as test app

Reply:
it's a convenience method to allow for easy tracking regardless of what level the customer is using these values.
e.g. - propositionItem, content-card type schema, ContentCard (a specific type of content-card schema), etc.